employees.rb

```diff
@@ -33,56 +36,66 @@ def initialize(full_name, id, office)
   end
 end
 
-def add_employee(employees)
+def print_help_add
   puts '[Add an employee]'
   puts 'a - to add a regular employee'
   puts 'e - to add a programmer'
   puts 'o - to add a office manager'
-  print 'who do you want to add? '
-  action = gets.chomp
+end
+
+def input_employee_details
   print 'Full name: '
   full_name = gets.chomp
   print 'ID: '
   id = gets.chomp
+  [full_name, id]
+end
 
+def add_employee(employees)
+  print_help_add
+  print 'who do you want to add? '
+  action = gets.chomp
   case action
-  when 'a' then employee = Employee.new(full_name, id)
-  when 'e' 
+  when 'a'
+    full_name, id = input_employee_details
+    employee = Employee.new(full_name, id)
+  when 'e'
+    full_name, id = input_employee_details
     print 'Languages: '
-    languages = gets.chomp.split(" ")
+    languages = gets.chomp.split(' ')
     employee = Programmer.new(full_name, id, languages)
   when 'o'
+    full_name, id = input_employee_details
     print 'Office: '
     office = gets.chomp
     employee = OfficeManager.new(full_name, id, office)
+  else
+    print_help_add
```

Comment:
This will only print out a message, the program would fail on line 75 because no employee variable was defined.
You could add an early return here, perhaps.